src/lib/upserts.ts

```diff
@@ -21,23 +21,20 @@ export async function upsertRegistration(
 }
 
 /**
- * Idempotent handler to ensure a domain entity for requested node exists in
- * the database. It inserts a domain entity if it does not exist. Otherwise,
- * just returns the existing domain entity from the db.
+ * A function to ensure a domain entity for requested node value exists in
```

Comment:
```suggestion
 * Ensure that some domain entity value (not necessarily the provided value) for the requested node exists in
```

Carefully rewording to reduce potential confusion that the node value in the database will be the node value provided.